src/app/pages/vm/vm-cards/vm-cards.component.ts

```diff
@@ -413,60 +399,111 @@ export class VmCardsComponent implements OnInit {
   }
   toggleForm(flipState, card, template){
     // load #cardBack template with code here
-    //console.log(flipState);
     card.template = template;
     card.isFlipped = flipState;
     card.lazyLoaded = !card.lazyLoaded;
-    var index = this.cards.indexOf(card);
+    const index = this.cards.indexOf(card);
     this.focusVM(index);
   }
 
   // toggles VM on/off
   toggleVmState(index){
-    let vm = this.cards[index];
+    const vm = this.cards[index];
     let eventName: string;
-    if (vm.state != 'running') {
-      eventName = "VmStart";
-    } else {
+    if (vm.state !== 'running') {
+      this.ws.call('vm.query', [[['id', '=', vm.id]]]).subscribe((res)=>{
+        for (const device of res[0].devices){
+          if(device.dtype === 'RAW' && device.attributes.boot){
+            this.raw_file_path = device.attributes.path;
+            this.raw_file_path_size = String(device.attributes.size);
+          }
+        }
+          if (res[0].vm_type === "Container Provider"){
+            this.dialogRef = this.matdialog.open(EntityJobComponent, { data: {title: 'Fetching RancherOS'}, disableClose: false});
+            this.dialogRef.componentInstance.progressNumberType = "nopercent";
+            this.dialogRef.componentInstance.setCall('vm.fetch_image', ['RancherOS']);
+            this.dialogRef.componentInstance.submit();
+            this.dialogRef.componentInstance.success.subscribe((sucess_res) => {
+              this.loader.open();
+              this.ws.call('vm.image_path', ['RancherOS']).subscribe((img_path)=>{
+                if(!img_path){
+                  this.dialog.Info('CHECKSUM MISMATCH', 'system checks failed to verify ISO, please try rebooting Virtual Machine');
```

Comment:
If for some reason the download has a checksum problem the VM won't start, but if you try to start it again, it will remove the image problem and download it again.

Maybe a better error message would be something like:
this.dialog.Info('CHECKSUM MISMATCH', 'system checks failed to verify ISO, please try to start again');